vertical-pod-autoscaler/pkg/admission-controller/Dockerfile

```diff
@@ -12,22 +12,27 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM --platform=$BUILDPLATFORM golang:1.23.3 AS builder
+FROM --platform=$BUILDPLATFORM golang:1.23.4 AS builder
 
-ENV GOPATH=/gopath/
-ENV PATH=$GOPATH/bin:$PATH
+WORKDIR /workspace
 
-COPY . /gopath/src/k8s.io/autoscaler/vertical-pod-autoscaler
-WORKDIR /gopath/src/k8s.io/autoscaler/vertical-pod-autoscaler
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+
+RUN go mod download
+
+COPY common common
+COPY pkg pkg
 
 ARG TARGETOS TARGETARCH
 
-RUN CGO_ENABLED=0 LD_FLAGS=-s GOARCH=$TARGETARCH GOOS=$TARGETOS go build -C pkg/admission-controller -mod vendor -o admission-controller-$TARGETARCH
+RUN CGO_ENABLED=0 LD_FLAGS=-s GOARCH=$TARGETARCH GOOS=$TARGETOS go build -C pkg/admission-controller -o admission-controller-$TARGETARCH
 
-FROM gcr.io/distroless/static:latest
+FROM gcr.io/distroless/static:nonroot
```

Comment:
Small nit - I wonder if we can split out this change into a different PR?

Then we can keep that change isolated (and it will be easier to spot during release if it causes wonky behavior).

Reply:
Sure, I can split that out.